lib/sparse/src/common/sparse_vector.rs

```diff
@@ -17,6 +19,95 @@ pub struct SparseVector {
     pub values: Vec<DimWeight>,
 }
 
+/// Same as `SparseVector` but with `DimOffset` indices.
+/// Meaning that is uses internal segment-specific indices.
+#[derive(Debug, PartialEq, Clone, Default, Serialize, Deserialize)]
+pub struct RemappedSparseVector {
+    /// indices must be unique
+    pub indices: Vec<DimOffset>,
+    /// values and indices must be the same length
+    pub values: Vec<DimWeight>,
+}
+
+/// Sort two arrays by the first array.
+fn double_sort<T: Ord + Copy, V: Copy>(indices: &mut [T], values: &mut [V]) {
```

Comment:
📝 Memo: Recently, I was trying to find an implementation of two-arrays-sort for Rust without additional memory allocation, but I couldn't find any.
There is a crate [permutation](https://crates.io/crates/permutation) that could be used, but it also allocates memory.

Shame on Rust since C++ apparently has it. https://stackoverflow.com/a/53785022

Reply:
Yeah, I didn't want to bring this additional crate just for this